### DISCUSSION TITLE
how to use where on doUpdate when using insertAll?

### DISCUSSION THREAD

**Discussion by rizalahmaddd** [simolus3/drift#3303] [Q&A]
If a conflict occurs, I want to compare the updatedAt column. How can I do that? I cannot access the new data in the DoUpdate callback.
`await batch((batch) {
            batch.insertAll(
              flags,
              data,
              onConflict: DoUpdate(
                (Flag old) => newData?
                where: (Flags old) => old.updatedAt.isSmallerOrEqual(newData.updatedAt?),
              ),
            );
          });`

thanks.

**Comment by simolus3** [comment]
You can use `DoUpdate.withExcluded` to have both rows available. To use data from the inserted row, just loop over the entries manually and return the data in the first function:

```dart
await batch((batch) {
  for (final entry in data) {
    batch.insert(
      flags,
      entry,
      onConflict: DoUpdate.withExcluded(
        (Flag old) => entry,
        where: (Flags old, Flags excluded) => old.updatedAt.isSmallerOrEqual(excluded.updatedAt)),
    );
  }
});
```
